#! /bin/sh -e

if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $pdir -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch $pdir -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# DP: support for GNU/FreeBSD

diff -ur readline4-4.3.old/support/shlib-install readline4-4.3/support/shlib-install
--- readline4-4.3.old/support/shlib-install	2001-08-23 16:38:25.000000000 +0200
+++ readline4-4.3/support/shlib-install	2003-07-09 03:05:15.000000000 +0200
@@ -95,7 +95,7 @@
 # Create symlinks to the installed library.  This section is incomplete.
 #
 case "$host_os" in
-*linux*|bsdi4*|*gnu*|darwin*|macosx*)
+*linux*|bsdi4*|gnu*|darwin*|macosx*|k*bsd*-gnu)
 	# libname.so.M -> libname.so.M.N
 	${echo} ${RM} ${INSTALLDIR}/$LINK2
 	if [ -z "$uninstall" ]; then
diff -ur readline4-4.3.old/support/shobj-conf readline4-4.3/support/shobj-conf
--- readline4-4.3.old/support/shobj-conf	2002-04-17 19:27:46.000000000 +0200
+++ readline4-4.3/support/shobj-conf	2003-07-09 03:34:14.000000000 +0200
@@ -96,8 +96,8 @@
 	SHLIB_LIBVERSION='$(SHLIB_LIBSUFF).$(SHLIB_MAJOR)'
 	;;
 
-# All versions of Linux or the semi-mythical GNU Hurd.
-linux*|gnu*)
+# GNU and variants
+linux*-*|gnu*-*|k*bsd*-gnu-*)
 	SHOBJ_CFLAGS=-fPIC
 	SHOBJ_LD='${CC}'
 	SHOBJ_LDFLAGS='-shared -Wl,-soname,$@'

--===============0156869087==--
